frontend/src/app/data-entry/agent-form/agent-form.component.ts

```diff
@@ -1,27 +1,25 @@
-import { Component } from '@angular/core';
-import { Breadcrumb } from '@shared/breadcrumb/breadcrumb.component';
-import { dataIcons } from '@shared/icons';
-import { DataEntryAgentGQL, DataEntryAgentQuery } from 'generated/graphql';
-import { map, Observable, switchMap } from 'rxjs';
-import { FormService } from '../shared/form.service';
+import { Component } from "@angular/core";
+import { Breadcrumb } from "@shared/breadcrumb/breadcrumb.component";
+import { dataIcons } from "@shared/icons";
+import { DataEntryAgentGQL, DataEntryAgentQuery } from "generated/graphql";
+import { map, Observable, switchMap } from "rxjs";
+import { ActivatedRoute } from "@angular/router";
 
 @Component({
     selector: 'lc-agent-form',
     templateUrl: './agent-form.component.html',
     styleUrls: ['./agent-form.component.scss'],
-    providers: [FormService],
```

Comment:
I guess you could write something like

```ts
public id$(route: ActivatedRoute): Observable<string> {
        return route.params.pipe(map((params) => params["id"]));
    }
```

but I'm not sure it's worth it.

Reply:
No, providing the ID in the service was something I did later, because there was now a service with that scope. It saves some code, but I don't really mind either way.

I just made a [similar comment](https://github.com/CentreForDigitalHumanities/lettercraft/pull/126#discussion_r1773050164). To reiterate:

- It's intentional that the service creates a separate instance for each component, because it's used to track a state local to the component.
- The "standard singleton pattern" is appropriate when services are stateless or track application-level states, but that's not the case here.

Currently, we only show one form at a time, so a singleton service instance could theoretically work, though I think it makes the application more error-prone. (E.g., you might see statuses bleeding over from one form to the next when the user is navigating.) Though in the mock-up, we also had this kind of off-canvas form:

![screenshot of lettercraft. the screen shows a form where the user is editing a lit of political regions. a sidebar is opened to edit details of a single region.](https://github.com/user-attachments/assets/1136448c-fcdc-430a-a5ec-a6f16df6e125)

where you may expect the sidebar (which is editing a related object) to have its own status bar.

Reply:
Alright, I see what you mean. (I did not want to suggest that it is superior to having multiple instantiations, but singleton/state-wide services _are_ most common, so I felt that a deviation from that pattern should be motivated.) I'll adjust this PR accordingly and put the component-scoped services back.